content/draft/general/keyboard_shortcuts.md

```diff
@@ -0,0 +1,111 @@
+---
+title: "Keyboard shortcuts spec"
+description: ""
+date: 2020-12-01T08:48:57+00:00
+lastmod: 2020-12-01T08:48:57+00:00
+draft: false
+images: []
+menu:
+  raw:
+    parent: "general"
+weight: 100
+toc: true
+---
+
+# Keyboard Shortcuts spec
+
+## Motivation
+
+Users often use keyboard shortcuts to more effectively navigate applications. This spec aims to document the keyboard shortcuts available as well as any preconditions.
+
+## Designs
+
+* [designs](https://www.figma.com/file/IPpvkpDWabBKJTeo6bFop0/Kuba%E2%8E%9CDesktop?node-id=355%3A82493)
+
+## Functional Requirements
+
+### Key definition
+
+The requirements will use the keys `CMD` and `OPT` which are defined for each platform as follows:
+
+- `CMD`
+    - macOS: `⌘` or `CMD`
+    - Windows: `Ctrl`
+    - Linux: `Ctrl`
+- `OPT`
+    - macOS: `⌥` or `OPT`
+    - Windows: `Alt`
+    - Linux: `Alt`
+
+### Keyboard shortcuts help
+
+- `CMD + /` MUST open a modal listing available key shortcuts
+- The Modal listing available keyshortcuts MUST display the correct keys depending on the platform, for example `CMD + C` on macOS and `Ctrl + C` on Windows.
+
+### Keyboard shortcuts
+
+#### Message focus
+
+- `TAB` - MUST trigger focus of the last message
+- `UP` - MUST focus on the previous message
+    - requirement: there is already a message in focus
+- `DOWN` - MUST focus on the previous message
```

Comment:
This should probably say "next" nessage

Reply:
This is only if a message is focused.  If (for example) an item in the L1 or L2 nav is focused then the interaction is different.  And the really hard bit to implementing correctly is what happens if the user presses the RIGHT or LEFT cursor arrows to jump from say the L1 to the L2 or from the L2 to the message body area.  This is difficult because it means we need to understand relative vertical positioning between these elements.  Which is why I'm suggesting we descope focus based navigation for now, and come back to this task after we've got the Community MVP launched.